bool turn_initiated
bool turn_completed
